packages/neo-one-utils/src/changeCase.ts

```diff
@@ -0,0 +1,145 @@
+      },
+    },
+    az: {
+      regexp: /[\u0130]/g,
+      map: {
+        İ: '\u0069',
+        I: '\u0131',
+        İ: '\u0069',
+      },
+    },
+    lt: {
+      regexp: /[\u0049\u004A\u012E\u00CC\u00CD\u0128]/g,
+      map: {
+        I: '\u0069\u0307',
+        J: '\u006A\u0307',
+        Į: '\u012F\u0307',
+        Ì: '\u0069\u0307\u0300',
+        Í: '\u0069\u0307\u0301',
+        Ĩ: '\u0069\u0307\u0303',
+      },
+    },
+  };
+  const lang = locale === undefined ? undefined : LANGUAGES[locale];
+
+  const nextStr = String(str);
+
+  if (lang) {
+    return nextStr.replace(lang.regexp, (m) => lang.map[m]).toLowerCase();
+  }
+
+  return nextStr.toLowerCase();
+};
+
+export const upperCase = (str: string, locale?: Locale): string => {
+  const LANGUAGES: Languages = {
+    tr: {
+      regexp: /[\u0069]/g,
+      map: {
+        i: '\u0130',
+      },
+    },
+    az: {
+      regexp: /[\u0069]/g,
+      map: {
+        i: '\u0130',
+      },
+    },
+    lt: {
+      regexp: /[\u0069\u006A\u012F]\u0307|\u0069\u0307[\u0300\u0301\u0303]/g,
+      map: {
+        i̇: '\u0049',
+        j̇: '\u004A',
+        į̇: '\u012E',
+        i̇̀: '\u00CC',
+        i̇́: '\u00CD',
+        i̇̃: '\u0128',
+      },
+    },
+  };
+
+  const lang = locale === undefined ? undefined : LANGUAGES[locale];
+
+  let nextStr = String(str);
+
+  if (lang) {
+    nextStr = str.replace(lang.regexp, (m) => lang.map[m]);
+  }
+
+  return nextStr.toUpperCase();
+};
+
+export const noCase = (str: string, locale?: Locale, replacement?: string): string => {
+  const nextReplacement = typeof replacement !== 'string' ? ' ' : replacement;
+
+  function replace(match: string, index: number, value: string): string {
+    if (index === 0 || index === value.length - match.length) {
+      return '';
+    }
+
+    return nextReplacement;
+  }
+
+  const nextStr = String(str)
+    // Support camel case ("camelCase" -> "camel Case").
+    .replace(CAMEL_CASE_REGEXP, '$1 $2')
+    // Support odd camel case ("CAMELCase" -> "CAMEL Case").
+    .replace(CAMEL_CASE_UPPER_REGEXP, '$1 $2')
+    // Remove all non-word characters and replace with a single space.
+    .replace(NON_WORD_REGEXP, replace);
+
+  // Lower case the entire string.
+  return lowerCase(nextStr, locale);
+};
+
+export const upperCaseFirstNative = (str: string, locale?: Locale): string => {
+  const nextStr = String(str);
+
+  return upperCase(nextStr[0], locale) + nextStr.substr(1);
+};
+
+export const camel = (value: string, locale?: Locale, mergeNumbers?: boolean): string => {
+  let result = noCase(value, locale);
+
+  // Replace periods between numeric entities with an underscore.
+  if (!mergeNumbers) {
+    result = result.replace(/ (?=\d)/g, '_');
+  }
+
+  // Replace spaces between words with an upper cased character.
+  return result.replace(/ (.)/g, (_m, $1) => upperCase($1, locale));
+};
+
+export const pascal = (value: string, locale?: Locale, mergeNumbers?: boolean): string =>
```

Comment:
You can do a similar refactor like I described above here.

A really good rule of thumb to have in general is functions should have *AT MOST* 2 arguments. Sometimes we get away with breaking that rule for short utility functions but its good practice to refactor any number of arguments >2 into 1 options object.